fast64_internal/f3d/f3d_gbi.py

```diff
@@ -2821,6 +2840,289 @@ def to_c(self, f3d, gfxFormatter):
         return staticData, dynamicData
 
 
+class BleedGraphics:
+    def __init__(self):
+        self.bleed_state = self.BleedTag.BleedNone
+
+    def bleed_fModel(self, fModel, fMeshes):
+        if not fModel.inline:
+            return
+        # walk fModel, no order to drawing is observed, so lastMat is not kept track of
+        for drawLayer, fMesh in fMeshes.items():
+            self.bleed_fmesh(fModel.f3d, fMesh, None, fMesh.draw, drawLayer)
+        self.clear_gfx_lists(fModel)
+
+    # clear the gfx lists so they don't export
+    def clear_gfx_lists(self, fModel):
+        for (fMaterial, texDimensions) in fModel.materials.values():
+            fMaterial.material.commands = None
+            fMaterial.revert = None
+        for fMesh in fModel.meshes.values():
+            for tri_list in fMesh.triangleGroups:
+                tri_list.triList.commands = None
```

Comment:
Yeah it is about the same amount of work but I was weighing options.
If I clear the gfx list, you will get exceptions in any place you don't do a None check, but it might be clearer in intention. So adding may this requires more checks in various places and possible issues if specific subclasses try to use their own write methods.
Clearing the command list makes it easy to check inside the GfxList as a central location, and maybe even carries over into cases where you don't clear the list but it has no commands anyway.
If it is better to do the gfx list itself I could easily do that though.

Reply:
I think its better to push the complexity upwards into the owning classes (`FMaterial` and `FTriGroup`). Even if you have to handle more specific cases, this lets you use `GfxList` for other purposes without having to worry about null checking on the commands.